host/crypto/bcrypt/random.c

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <openenclave/internal/raise.h>
+#include <openenclave/internal/random.h>
+#include "bcrypt.h"
+
+oe_result_t oe_random_internal(void* data, size_t size)
+{
+    oe_result_t result = OE_UNEXPECTED;
+
+    /* For consistency with OpenSSL, keep this to int max */
+    if (size > OE_INT_MAX)
+        return OE_INVALID_PARAMETER;
+
+    NTSTATUS status = BCryptGenRandom(
+        NULL, data, (ULONG)size, BCRYPT_USE_SYSTEM_PREFERRED_RNG);
+
+    if (!BCRYPT_SUCCESS(status))
+        OE_RAISE_MSG(
+            OE_CRYPTO_ERROR, "BcryptGenRandom failed (err=%#x)\n", status);
```

Comment:
Is %#x specifier better than the 0x%x specifiers used for other OE_CRYPTO_ERROR instances in say enclave/key.c?

Reply:
It's in the realm of personal taste. IIRC, `%#x` will omit the '0x' if the value is 0 which I find more readable. It also keeps the casing of the prefix consistent with the casing of the hex value.